noir-projects/aztec-nr/aztec/src/macros/functions/call_interface_stubs.nr

```diff
@@ -0,0 +1,243 @@
+use crate::macros::utils::{
+    add_to_field_slice, AsStrQuote, compute_fn_selector, is_fn_private, is_fn_view,
+};
+use std::meta::{type_of, unquote};
+
+comptime global FROM_FIELD: TypedExpr = {
+    let from_field_trait = quote { protocol_types::traits::FromField }.as_trait_constraint();
+    let function_selector_typ =
+        quote { protocol_types::abis::function_selector::FunctionSelector }.as_type();
+    function_selector_typ.get_trait_impl(from_field_trait).unwrap().methods().filter(|m| {
+        m.name() == quote { from_field }
+    })[0]
+        .as_typed_expr()
+};
+
+comptime global SERIALIZED_ARGS_SLICE_NAME: Quoted = quote { serialized_args };
+
+pub comptime fn stub_fn(f: FunctionDefinition) -> Quoted {
```

Comment:
This function was originally in `interfaces.nr` but I have completely rewrote it because it was impossible to read. In hindsight I shouldn't have moved it in the same PR. Sorry for the bad diff.